lib/puppet/type/file/source.rb

```diff
@@ -273,18 +261,18 @@ def copy_source_value(metadata_method)
       end
     end
 
-    def each_chunk_from
-      if Puppet[:default_file_terminus] == :file_server
-        yield content
+    def each_chunk_from(&block)
+      if Puppet[:default_file_terminus] == :file_server && scheme == 'puppet' && (uri.host.nil? || uri.host.empty?)
+        chunk_file_from_disk(metadata.path, &block)
       elsif local?
-        chunk_file_from_disk { |chunk| yield chunk }
+        chunk_file_from_disk(full_path, &block)
       else
-        chunk_file_from_source { |chunk| yield chunk }
+        chunk_file_from_source(&block)
       end
     end
 
-    def chunk_file_from_disk
-      File.open(full_path, "rb") do |src|
+    def chunk_file_from_disk(local_path)
```

Comment:
should this be `def chunk_file_from_disk(local_path, &block)`?

Reply:
same question for `def chunk_file_from_source`. Though, is it implied if you just pass it a block? I feel like it is, and I'm just forgetting

Reply:
Yeah, adding an explicit block argument is only required if the method wants to refer to the block by name, such as when passing the block along to another method. For example, `each_chunk_from(&block)` takes an explicit block so that it can call `chunk_file_from_source(&block)` passing the block along.

An explicit block is not required in cases where the method being called does a `yield`. In that case it's faster to pass an implicit block to the method, as ruby doesn't have to convert the implicit block to a Proc. The latter can be a performance issue, but it doesn't matter here due to file I/O.

I thought there was a rubocop rule about explicit/implicit blocks, but I can't find it. Happy to add the explicit block arguments if it makes things clearer?

Reply:
Ah, okay, that makes sense. I think it's fine to leave it implicit (as is) since that's how it previously was